docs/proposals/oras-attach-multi-platform.md

```diff
@@ -0,0 +1,84 @@
+
+Related GitHub issue: https://github.com/oras-project/oras/issues/1531
+
+When a multi-platform image is created using the [OCI image index format](https://github.com/opencontainers/image-spec/blob/v1.1.1/image-index.md), users may want to attach a file and propagate it to platform-specific images. However, the current [oras attach](https://oras.land/docs/commands/oras_attach) command only allows users to attach a file to the image index or a single platform-specific image. Several scenarios require attaching a file as a referrer to both a multi-platform image and platform-specific images.
+
+## Scenarios
+
+### Attach an End-of-Life (EoL) Annotation as a Referrer to a Multi-Platform Image
+
+#### Scenario A: attach a referrer to a platform-specific image to upward propagate to its parent index
+
+A security engineer, Alice, wants to use annotations to store EoL metadata to indicate that an image is no longer valid. Consider a multi-platform image `demo/alpine:a1a1` with multiple platform-specific images. When a vulnerability is detected in a platform-specific image `demo/alpine:b1b1`, it is patched, generating a new digest. The outdated image is marked as invalid using an EoL annotation:
+
+```json
+"vnd.demo.artifact.lifecycle.end-of-life.date": "2025-03-20T01:20:30Z"
+```
+
+Set environment variables:
+
+```
+ARTIFACT_TYPE=application/vnd.demo.artifact.lifecycle
+ANNOTATION_KEY=vnd.demo.artifact.lifecycle.end-of-life.date
+ANNOTATION_VALUE=2025-03-20T01:20:30Z
+```
+
+Since the patched image has a new digest, the parent image index also receives an updated digest. When a platform-specific image is marked as EoL, dependent services stop using it, and vulnerability scanning tools can recognize it as deprecated through the lifecycle metadata.
+
+![multi-arch image](./img/oras-attach-EoL.drawio.svg)
+
+After patching `demo/alpine:b1b1`, Alice has to manually retrieve the new digests and run `oras attach` twice: once for the old digest of the parent image index and once for the platform-specific image manifest. If multiple platform-specific images require updates, she has execute multiple commands.
+
+```console
+demo/alpine:a1a1 (image index)
+-> demo/alpine:b1b1  <-- VULNERABLE
+-> demo/alpine:c1c1
+```
+
+After patching, Alice attaches the EoL annotation to the outdated image index and platform-specific image:
+
+```sh
+oras attach $registry/demo/alpine:a1a1 --artifact-type $ARTIFACT_TYPE --annotation $ANNOTATION_KEY=$ANNOTATION_VALUE
+oras attach $registry/demo/alpine:a1a1 --artifact-type $ARTIFACT_TYPE --annotation $ANNOTATION_KEY=$ANNOTATION_VALUE --platform linux/amd64
+```
+
+Resulting image structure:
+
+```console
+demo/alpine:a1a1 (image index) <-- VULNERABLE, EoL attached
+-> demo/alpine:b1b1 (linux/amd64)  <-- VULNERABLE, EoL attached
+-> demo/alpine:c1c1 (linux/arm64)
+```
+
+A new image index is created:
+
+```console
+demo/alpine:z1z1 (image index)
+-> demo/alpine:r1r1 (linux/amd64)
+-> demo/alpine:c1c1 (linux/arm64)
+```
```

Comment:
nit: So `b1b1` now becomes `r1r1` and `a1a1` now becomes `z1z1`? Does it mean that the image tags are updated after the images are patched? It might be strange as tags are usually reused while digests change. 
Would it be better to use `demo/alpine@sha256:a1a1` to represent the images?